File: code/snippets/swap-chars/swap.php
<?php
# strategy: use built-in 
# strtr php function
function swapChars($str, $a, $b = null) {
  $swappedString = strtr($str, 
    $b === null ? $a : 
      (strlen($a) === 1 ? $a . $b : $a), 
    $b === null ?
      (strlen($a) === 2 ? 
      strrev($a) :
      str_shuffle($a)) : 
      (strlen($b) === 1 ? $b . $a : $b)
  );
  return $swappedString;
}

# prints @rolling_codes
echo swapChars('@rodding_coles', 'dl');
# prints 666 in binary
echo swapChars('1111110101100101', 
               '0', '1');
?>

File: code/snippets/swap-chars/swap.py
# strategy: use built-in 
# maketrans method
def swap_chars(string, a, b):
  swapped_string = string.translate(
    str.maketrans(a + b, b + a))
  return swapped_string

# prints @rolling_codes
print(
  swap_chars('rossing_codel', 'l', 's'))
# prints 42 in binary
print(
  swap_chars(
    '1111111111010101', '0', '1'))
